download_test_vectors.sh

```diff
@@ -2,16 +2,58 @@
 
 set -eu
 
-source scripts/download_functions.sh
+FLAVOURS=(
+	"general"
+	"minimal"
+	"mainnet"
+)
 
-dl_version v0.8.3
-dl_version v0.9.0
+dl_version() {
+	[[ -z "$1" ]] && { echo "usage: dl_version() vX.Y.Z"; exit 1; }
+	version="$1"
 
-echo "Ignore the warnings \"unknown extended header keyword 'SCHILY.{dev,ino,nlink}'\" on Linux."
-# tar: Ignoring unknown extended header keyword 'SCHILY.dev'
-# tar: Ignoring unknown extended header keyword 'SCHILY.ino'
-# tar: Ignoring unknown extended header keyword 'SCHILY.nlink'
-echo "Those are due to the test vectors being packed with OSX BSD tar."
+	mkdir -p "tarballs/${version}"
+	pushd "tarballs/${version}" >/dev/null
+	for flavour in "${FLAVOURS[@]}"; do
+		if [[ ! -e "${flavour}.tar.gz" ]]; then
+			echo "Downloading: ${version}/${flavour}.tar.gz"
+			curl --location --remote-name --silent --show-error \
+				"https://github.com/ethereum/eth2.0-spec-tests/releases/download/${version}/${flavour}.tar.gz" \
+				|| {
+					echo "Curl failed. Aborting"
+					rm -f "${flavour}.tar.gz"
+					exit 1
+				}
+		fi
+	done
+	popd >/dev/null
+}
+
+unpack_version() {
+	[[ -z "$1" ]] && { echo "usage: unpack_version() vX.Y.Z"; exit 1; }
+	version="$1"
+
+	dl_version "$version"
+
+	# suppress warnings when unpacking with GNU tar an archive created with BSD tar (probably on macOS)
+	EXTRA_TAR_PARAMS=""
+	tar --version | grep -qi 'gnu' && EXTRA_TAR_PARAMS="--warning=no-unknown-keyword"
```

Comment:
good call, I didn't see how to have the --warning=no-unknown-keyword for GNU and nothing for BSD in a nice manner